halo2_backend/src/plonk/prover.rs

```diff
@@ -563,14 +643,19 @@ impl<
             )
             .collect();
 
-        // Evaluate the h(X) polynomial
+        // 7. Evaluate the h(X) polynomial -----------------------------------------------------------
+
+        // Obtain challenge for keeping all separate gates linearly independent
+        // [TRANSCRIPT-13]
+        let y: ChallengeY<_> = self.transcript.squeeze_challenge_scalar();
+
         let h_poly = pk.ev.evaluate_h(
```

Comment:
I would change this to:
```
let g_poly = pk.ev.calculate_g(
```

Please if somebody knows why this was called `evaluate_h`, can they explain it to me?

Reply:
hm, but in the code this variable holds the polynomial that contains `gate_1(X) + alpha * gate_2(X) + alpha^2 * gate_3(X) + ...`, which is not the vanishing polynomial right?